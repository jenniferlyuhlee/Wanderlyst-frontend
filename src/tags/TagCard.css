.TagCard{
    overflow: hidden;
}

.Tag-bg{
    position: relative;
    width: 100%;
    height: 15vw; /* Adjust the height as needed */
    background-size: cover;
    background-position: center;
    background-repeat: no-repeat;
    transition: transform 0.4s ease-in-out;
}

.Tag-bg:hover{
    transform: scale(1.05);
}

@media (max-width: 1000px){
    .Tag-bg{
        height: 30vw;
    }
}

.TagCard-overlay{
    position: absolute;
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    height: 100%;
    color: rgba(0, 0, 0, 0);
    transition: background-color 0.4s ease-in-out, color 0.4s ease-in-out;
}

.TagCard-overlay:hover{
    background-color: rgba(0, 0, 0, 0.5);
    color: white;
}

/* Dynamic classes for each tag background */

/* Image sourced from:
https://media.worldnomads.com/responsibletravel/foodgetty.jpg */
.tag-bg-Foodie{
    background-image: url("../assets/tag-images/foodie.jpg");
}
/* Image sourced from: 
https://www.magnificentworld.com/wp-content/uploads/2020/07/Petra-Travel-Guide.jpg */
.tag-bg-Explorer{
    background-image: url("../assets/tag-images/explorer.jpg");
}
/* Image sourced from: 
https://www.travelmoneyoz.com/sites/v1.travelmoneyoz.com/files/fcl-blog/Shopping%20in%20Europe.jpg */
.tag-bg-Shopaholic{
    background-image: url("../assets/tag-images/shopaholic.jpg");
}
/* Image sourced from: 
https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTJAcwcAlB303gRJ-q2XiDclKsbAOHoA-lJ5Q&s */
.tag-bg-Naturalist{
    background-image: url("../assets/tag-images/naturalist.png");
}
/* Image sourced from:
https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTA1QntsFS7YAxUvy-pWxAPV5j6aejGrQ8WUA&s */
.tag-bg-Daredevil{
    background-image: url("../assets/tag-images/daredevil.jpg");
}
/* Image sourced from: 
https://assets.cntraveller.in/photos/66683fdbf2c9eddc5e47e147/16:9/w_1280,c_limit/GettyImages-1388400714%20(1).jpg */
.tag-bg-Flâneur{
    background-image: url("../assets/tag-images/flâneur.jpg");
}
/* Image sourced from: 
https://timenightclub.com/wp-content/uploads/2024/06/orange-county-nightlife-clubbing-near-me-costa-mesa-ca-1-1024x576.webp */
.tag-bg-Reveller{
    background-image: url("../assets/tag-images/reveller.jpeg");
}
/* Image sourced from: 
https://static01.nyt.com/images/2019/11/29/smarter-living/00sl-museums/merlin_156969357_5c37f256-0b6d-400d-bfba-cabd3e6c4dfd-superJumbo.jpg */
.tag-bg-Aesthete{
    background-image: url("../assets/tag-images/aesthete.jpg");
}
/* Image sourced from: 
https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQ0UJUoR-pIhad6jpHdpWBBC_s714RW7x7g8Q&s */
.tag-bg-Relaxer{
    background-image: url("../assets/tag-images/relaxer.jpg");
}
/* Image sourced from:
https://i.natgeofe.com/n/fba4cbb4-6a7f-4a4d-9e32-9f691cef8856/higts_southafrica_8502902-Edit_HR_square.jpg */
.tag-bg-Photographer{
    background-image: url("../assets/tag-images/photographer.jpg");
}

