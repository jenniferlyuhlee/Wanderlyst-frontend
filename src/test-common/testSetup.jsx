import React, { useState } from "react";
import UserContext from "../auth-user/UserContext";
import WanderlystApi from "../utils/api";
import '@testing-library/jest-dom';

const demoUser = {
  username: "testadmin", 
  firstName: "testfirst",
  lastName: "testlast",
  location: "testloc",
  bio: "testbio",
  profilePic: null,
  isAdmin: true,
};

const demoTags = [
  {id: 1, name: 'Foodie', description: 'Savor the world one bite at a time. Dive into a journey packed with delicious local cuisines, hidden culinary gems, and must-try dishes that will leave your taste buds tingling.'},
  {id: 2, name: 'Explorer', description: 'Step off the beaten path and uncover the unknown. From ancient ruins to vibrant cityscapes, let your curiosity guide you through unforgettable experiences.'}
]

const UserProvider = ({ children, 
                        currUser = demoUser, 
                        setCurrUser, 
                        tags = demoTags,
                        logout,
                        hasLikedItin,
                        toggleLike }) => {
  const [user, setUser] = useState(currUser);

  return (
    <UserContext.Provider value={{ currUser: user, 
                                  setCurrUser: setUser, 
                                  tags, logout, 
                                  hasLikedItin, toggleLike}}>
      {children}
    </UserContext.Provider>
  );
};

export { UserProvider };