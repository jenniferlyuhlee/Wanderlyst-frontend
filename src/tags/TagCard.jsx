import React from "react";
import { Link } from "react-router-dom";
import "./TagCard.css"

/** TagCard Component
 * props: tag: {name, description}
 */
function TagCard({tag}){
    return(
        <Link to={`/tags/${tag.name}`} className = "TagCard">
            <div className = {`Tag-bg tag-bg-${tag.name}`}>
                <div className = "TagCard-overlay">
                    <h4>{tag.name}</h4>
                </div>
            </div>
        </Link>
    )
}

export default TagCard;