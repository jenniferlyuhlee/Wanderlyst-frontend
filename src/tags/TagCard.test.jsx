import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { UserProvider } from "../test-common/testSetup";
import { MemoryRouter, Routes, Route} from "react-router-dom";
import TagCard from "./TagCard";
import { expect, describe, test} from "vitest";

const testTag = {
    name: "testTag",
    description: "Tag for testing."
}

describe("TagCard component", () => {
    test("renders without crashing", () => {
        render(
            <MemoryRouter>
                <UserProvider>
                    <TagCard tag={testTag}/>
                </UserProvider>
            </MemoryRouter>)
    });

    test("matches snapshot", () => {
        const {getByText, asFragment} = render(
            <MemoryRouter>
                <UserProvider>
                    <TagCard tag={testTag}/>
                </UserProvider>
            </MemoryRouter>)

        expect(asFragment()).toMatchSnapshot();
        expect(getByText("testTag")).toBeInTheDocument();
    });

    test("clicking card link works", () => {
        const {container, getByText} = render(
            <MemoryRouter>
                <UserProvider>
                    <Routes>
                        {/* Define route for the TagCard */}
                        <Route path="/" element={<TagCard tag={testTag} />} />
                        {/* Define the route you expect to navigate to */}
                        <Route path="/tags/:name" element={<div>Tag Details Page</div>} />
                    </Routes>
                </UserProvider>
            </MemoryRouter>)

        const cardLink = container.querySelector(".TagCard");
        fireEvent.click(cardLink);
        expect(getByText("Tag Details Page")).toBeInTheDocument();
    });
});