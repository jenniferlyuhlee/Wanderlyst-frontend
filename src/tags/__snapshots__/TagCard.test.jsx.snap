// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TagCard component > matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="TagCard"
    href="/tags/testTag"
  >
    <div
      class="Tag-bg tag-bg-testTag"
    >
      <div
        class="TagCard-overlay"
      >
        <h4>
          testTag
        </h4>
      </div>
    </div>
  </a>
</DocumentFragment>
`;
